### PR TITLE
RKE log metadata

### PR DESCRIPTION
**Pull request by paynejacob**
PROBLEM:

RKE logs are collected but there is no indication of what service they came from.  This makes it impossible to write a filter like "only etcd logs"

SOLUTION:
Aggregate all the rke logs with a 2nd fluentbit container and append their filenames, this allows use to filter on filenames and even container ids.

NOTES:

The `logging-containers-rke.yaml` was removed because it is redundant, all rke containers are capture by the new aggregator and all other containers are capture by the root logging.

### CODE REVIEW

**Comment by paynejacob** [comment]
https://github.com/rancher/rancher/issues/29410

**Comment by paynejacob** [comment]
Example output

```

      {
        "_index" : "rke",
        "_type" : "_doc",
        "_id" : "-McCJHUB2K2LlOa4RDXC",
        "_score" : 7.742002,
        "_source" : {
          "log" : "rke: [1602624548.103580, {\"filename\":\"/var/lib/rancher/rke/log/etcd_09a3c7dfb72717fe26bf7dc02556a33805f1b2402d779d599a089eec1a55b3e6.log\",\"log\":\"{\\\"log\\\":\\\"2020-10-12 21:46:04.636259 I | embed: peerTLS: cert = /etc/kubernetes/ssl/kube-etcd-104-131-126-81.pem, key = /etc/kubernetes/ssl/kube-etcd-104-131-126-81-key.pem, trusted-ca = /etc/kubernetes/ssl/kube-ca.pem, client-cert-auth = true, crl-file = \\\\n\\\",\\\"stream\\\":\\\"stderr\\\",\\\"time\\\":\\\"2020-10-12T21:46:04.63638722Z\\\"}\"}]"
        }
      }
```

**Review by nickgerace — APPROVED** [review]

**Review by paynejacob — COMMENTED** [review]
Need to add NIck's toleration and node selector changes.

**Review by brendarearden — APPROVED** [review]
Looks 🔥

**Review by nickgerace — APPROVED** [review]

File: packages/rancher-logging/overlay/templates/loggings/rke/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        Log_Level         info
+        Parsers_File      parsers.conf
+
+    [INPUT]
+        Tag               rke
+        Name              tail
+        Path_Key          filename
+        Parser            json
+        DB                /tail-db/tail-containers-state.db
+        Mem_Buf_Limit     5MB
```
Comment (cbron) [inline packages/rancher-logging/overlay/templates/loggings/rke/configmap.yaml:20]: Should this be editable ?

File: packages/rancher-logging/overlay/templates/loggings/rke/logging-rke.yaml [hunk 2]
```diff
@@ -14,10 +14,11 @@ spec:
       tag: {{ .Values.images.fluentbit.tag }}
     inputTail:
       Tag: "rke"
-      Path: "/rke/*.log"
```
Comment (cbron) [inline packages/rancher-logging/overlay/templates/loggings/rke/logging-rke.yaml:17]: What did this path previously do ?

File: packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        - name: outdir
+          hostPath:
+            path: /var/lib/rancher/logging/
+        - name: containers
+          hostPath:
+            path: /var/lib/docker/containers/
+        - name: config
+          configMap:
+            name: "{{ .Release.Name }}-rke"
+      {{- $total_tolerations := concat (.Values.tolerations) (.Values.fluentbit_tolerations) }}
```
Comment (cbron) [inline packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml:43]: Do we include fluentbit tolerations here if its a fluentd ?

File: packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+              name: containers
+            - mountPath: /fluent-bit/etc/
+              name: config
+      volumes:
+        - name: indir
+          hostPath:
+            path: /var/lib/rancher/rke/log/
+        - name: outdir
+          hostPath:
+            path: /var/lib/rancher/logging/
```
Comment (cbron) [inline packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml:36]: It's ok that both fluentD's output to same directory ?

File: packages/rancher-logging/overlay/templates/loggings/rke/configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-rke
+  labels:
+{{ include "logging-operator.labels" . | indent 4 }}
+data:
+  fluent-bit.conf: |
+    [SERVICE]
+        Log_Level         info
```
Comment (cbron) [inline packages/rancher-logging/overlay/templates/loggings/rke/configmap.yaml:11]: Should anything in this file be customizable ? Log level seems like something that should be.

**Review by cbron — COMMENTED** [review]
So the new architecture is that when you turn logging on in an rke cluster you get: 
* fluentd (from logging)
* fluentbit (from logging)
* 2nd fluentD (from rke daemonset)

File: packages/rancher-logging/overlay/templates/loggings/rke/logging-rke.yaml (hunk 2 above)
Comment (paynejacob) [reply packages/rancher-logging/overlay/templates/loggings/rke/logging-rke.yaml:17]: previously this was how we collected the rke logs but it yielded no meta data, this pr replaced it.

File: packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml (hunk 3 above)
Comment (paynejacob) [reply packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml:43]: The ds is a fluentbit

File: packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml (hunk 4 above)
Comment (paynejacob) [reply packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml:36]: yes the files are different names

File: packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml (hunk 3 above)
Comment (cbron) [reply packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml:43]: Oh, I was looking at line 19, that still says fluentD, the image is using fluentbit.

Comment (paynejacob) [reply packages/rancher-logging/overlay/templates/loggings/rke/daemonset.yaml:43]: opps will fix

File: packages/rancher-logging/rancher-logging.patch [hunk 6]
```diff
@@ -144,3 +144,7 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-logging/charts-original/values
 +global:
 +  cattle:
 +    systemDefaultRegistry: ""
++
++rke_fluentbit:
++  log_level: "info"
++  mem_buffer_limit: "5MB"
```
Comment (cbron) [inline packages/rancher-logging/rancher-logging.patch:150]: This could go above under additional logging sources ?

Comment (paynejacob) [reply packages/rancher-logging/rancher-logging.patch:150]: that is a good idea

File: packages/rancher-logging/rancher-logging.patch [hunk 7]
```diff
@@ -111,6 +111,9 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-logging/charts-original/values
 +additionalLoggingSources:
 +  rke:
 +    enabled: false
++    rke_fluentbit:
```
Comment (cbron) [inline packages/rancher-logging/rancher-logging.patch:114]: The double rke is redundant
```suggestion
+    fluentbit:
```

**Review by cbron — COMMENTED** [review]
Think this makes sense, one more change. I forget who originally made the issue but can we get them to test these changes

**Review by cbron — APPROVED** [review]

**Pull request merged**